service/history/tasks/category.go

```diff
@@ -115,7 +114,7 @@ var (
 
 // NewCategory creates a new Category and register the created Category
 // Registered Categories can be retrieved via GetCategories() or GetCategoryByID()
-// NewCategory panics when a Category with the same ID has already been registered
+// NewCategory does nothing if a Category with the same ID is already registered
```

Comment:
Why remove the panic logic? NewCategory() for the same category multiple times?

There's no way the system can recover from duplicated ID for different task categories. If we allow this case, then tasks of different category may end up stored in the same history task queue and data got corrupted. 

This file may not contain all task category definitions, so I think we need this check to prevent developer from reusing categoryIDs.

Reply:
FWIW this is a draft diff, and I was just testing things